### PR TITLE
Fbin account management

### PR DESCRIPTION
**Pull request by Fbin243**

### CODE REVIEW

**Comment by Fbin243** [comment]
I haven't finish the account management on this pull request

File: src/engine/keypair.ts [hunk 1]
```diff
@@ -12,27 +16,74 @@ export const createSeedPhrase = () => {
...
+
+  // Store the initial seed phrase
+  const storedSeedPhrase = {
+    type: StoredAccountType.SeedPhrase,
+    name: "Wallet 1",
+    seedPhrase: seedPhrase,
+    privateKeys: [
+      {
+        type: StoredAccountType.PrivateKey,
+        name: "Account 1.1",
```
Comment (devtronghieu) [inline src/engine/keypair.ts:30]: I think user don't want to understand what is `1.1`

File: src/engine/keypair.ts [hunk 2]
```diff
@@ -12,27 +16,74 @@ export const createSeedPhrase = () => {
...
+    activeWalletIndex: 0,
+    activeKeypairIndex: 0,
   };
 };
 
-export const deriveKeypair = async (password: string, pathIndex: number) => {
-  const encryptedSeedPhrase = await getEncryptedSeedPhrase();
-  if (!encryptedSeedPhrase) {
-    throw new Error("Seed phrase not found");
+export const deriveKeypairAndName = async (password: string) => {
```
Comment (devtronghieu) [inline src/engine/keypair.ts:53]: help me to rename the `password` arg to `hashedPassword`

File: src/engine/storage/connection.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import IDBPouch from "pouchdb-adapter-idb";
 import PouchDB from "pouchdb-core";
 
-let db: PouchDB.Database;
+let db: PouchDB.Database | null;
```
Comment (devtronghieu) [inline src/engine/storage/connection.ts:4]: you may want to add its default value to `null` also
`let db: PouchDB.Database | null = null`

File: src/icons/HideBalance.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { FC } from "react";
+
+import { IconProps } from "./types";
+
+export const HideBalance: FC<IconProps> = ({ size = 16, color = "#EF5385" }) => {
```
Comment (devtronghieu) [inline src/icons/HideBalance.tsx:5]: name it `HideBalanceIcon` (although it is not really an icon lol)

File: src/icons/HideBalance.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      <span className={i == 0 ? "ms-2 me-0.5" : "m-0.5"} key={i}>
+        <svg width={size} height={size} viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+          <rect width={size} height={size} rx="8" fill={color} />
+        </svg>
+      </span>,
+    );
+  }
+
+  return <div className="flex items-center text-2xl font-semibold text-center text-primary-400 mt-2">$ {svgs}</div>;
+};
```
Comment (devtronghieu) [inline src/icons/HideBalance.tsx:18]: you may try this trick

```tsx
{Array.from({ length: 10 }).map((_, index) => (
  <span className={i == 0 ? "ms-2 me-0.5" : "m-0.5"} key={i}>
      <svg width={size} height={size} viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
        <rect width={size} height={size} rx="8" fill={color} />
      </svg>
  </span>
))}
```

File: src/screens/Home/index.tsx [hunk 6]
```diff
@@ -10,31 +13,117 @@ import TransactionResult from "@screens/Result";
...
+  HideEyeIcon,
+  SettingIcon,
+  ShowEyeIcon,
+  SwapIcon,
+} from "@/icons/index";
 
 import Collections from "./Collections";
 
+interface AddrListItem {
+  //Must update with Logic
```
Comment (devtronghieu) [inline src/screens/Home/index.tsx:40]: what is the status of this comment now? have you updated it with the logic or not?

File: src/screens/Home/index.tsx
Comment (devtronghieu) [inline src/screens/Home/index.tsx:1]: this `file` is giant, please break it down to smaller files (by logic or by components, or both)

File: src/screens/Send/collectible.tsx
Comment (devtronghieu) [inline src/screens/Send/collectible.tsx:1]: I think it should be `Collectible.tsx`

File: src/screens/Send/token.tsx
Comment (devtronghieu) [inline src/screens/Send/token.tsx:1]: this file also, `PascalCase` for component file please (except the `index.tsx`)

File: src/screens/Settings/ManageAccountsBottomSheet.tsx [hunk 7]
```diff
@@ -1,5 +1,49 @@
...
+            jsx = account.privateKeys.map((privateKey, keypairIndex) => (
+              <SettingAccount
+                title={privateKey.name}
+                value={`$${formatCurrency(privateKey.lastBalance)}`}
+                onClick={() => onItemClick(walletIndex, keypairIndex, StoredAccountType.SeedPhrase, account.seedPhrase)}
+                hasIcon={false}
+                key={privateKey.privateKey}
+              />
+            ));
+            return jsx;
```
Comment (devtronghieu) [inline src/screens/Settings/ManageAccountsBottomSheet.tsx:32]: you can return it directly, don't need a `jsx` var to hold the logic

**Review by devtronghieu — APPROVED** [review]

File: src/engine/accounts/index.ts [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+
+  // Active account is private key, change it to the first seed phrase
+  if (activeWallet.type === StoredAccountType.PrivateKey) {
+    activeWalletIndexTemp = accounts.findIndex((acc) => acc.type === StoredAccountType.SeedPhrase);
+    activeWallet = accounts[activeWalletIndexTemp];
+    await upsertActiveIndex(PouchID.activeWalletIndex, activeWalletIndexTemp);
+  }
+
+  activeWallet = activeWallet as StoredSeedPhrase;
+  const privateKeys = activeWallet.privateKeys;
```
Comment (ltminhthu) [inline src/engine/accounts/index.ts:49]: why an active wallet has many private keys?

**Pull request merged**